app/lib/features/chat/widgets/rooms_list.dart

```diff
@@ -11,25 +13,185 @@ import 'package:flutter_svg/flutter_svg.dart';
 import 'package:go_router/go_router.dart';
 
 final bucketGlobal = PageStorageBucket();
-// interface providers
-final _searchToggleProvider = StateProvider.autoDispose<bool>((ref) => false);
 
-class RoomsListWidget extends ConsumerWidget {
+class RoomsListWidget extends ConsumerStatefulWidget {
   static const roomListMenuKey = Key('room-list');
 
-  const RoomsListWidget({
-    super.key = roomListMenuKey,
-  });
+  const RoomsListWidget({super.key = roomListMenuKey});
 
   @override
-  Widget build(BuildContext context, WidgetRef ref) {
+  ConsumerState<ConsumerStatefulWidget> createState() =>
+      _RoomsListWidgetState();
+}
+
+class _RoomsListWidgetState extends ConsumerState<RoomsListWidget> {
+  final ScrollController controller = ScrollController();
+  final TextEditingController searchTextController = TextEditingController();
+  final FocusNode searchFocus = FocusNode();
+
+  bool _isSearchVisible = false;
+
+  @override
+  void initState() {
+    super.initState();
+    controller.addListener(_onScroll);
+    searchTextController.text =
+        ref.read(roomListFilterProvider).searchTerm ?? '';
+  }
+
+  @override
+  void didUpdateWidget(RoomsListWidget oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    searchTextController.text =
+        ref.read(roomListFilterProvider).searchTerm ?? '';
+  }
+
+  void _onScroll() {
+    final topPosition = controller.position.pixels <= 0;
+    final outOfRange = controller.position.outOfRange;
+    final offset = controller.offset;
+    if (topPosition && outOfRange && offset <= -80) {
+      setState(() {
+        _isSearchVisible = true;
+      });
+    } else if (!topPosition && !outOfRange) {
+      setState(() {
+        _isSearchVisible = false;
+      });
+    }
+  }
+
+  Widget roomListTitle(BuildContext context) {
+    String title = AppLocalizations.of(context)!.chat;
+
+    if (ref.watch(hasRoomFilters)) {
+      String subFiltered = 'chats';
+      switch (ref
```

Comment:
keep conditionals simpler for better code reading ;)
```suggestion
    final selectionValue = ref.watch(roomListFilterProvider.select((value) => value.selection)));
      switch (selectionValue){
```